### PR TITLE
Translatebooknames

### PR DESCRIPTION
**Pull request by Ashersam**
Worked on #147  created UI based translation of book names. please review the code.

### CODE REVIEW

File: src/components/BookEditList.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    };
+    const onChange = (event, value) => {
+        setUpdatedValue(event.target.value)
+    }
+
+    const updateBooks = () => {
+        if (updatedValue !== "") {
+            AutographaStore.UpdatedBookName = updatedValue;
+            db.get('001', function (err, doc) {
```
[the remark this answers is not in the document]
Comment (Ashersam) [reply src/components/BookEditList.js]: Changed Name

File: src/components/BookEditList.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                if (err) {
+                    return console.log(err);
+                } else {
+                    doc.books.splice(AutographaStore.RequiredIndex, 1, AutographaStore.UpdatedBookName)
+                }
+                db.put(doc).then((response) => {
+                    console.log("updated", response)
```
[the remark this answers is not in the document]
Comment (Ashersam) [reply src/components/BookEditList.js]: cleaned test consoles

File: src/components/BookEditList.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import ArrowRightAltIcon from '@material-ui/icons/ArrowRightAlt';
+import { TextField, RaisedButton } from 'material-ui';
+import RestoreIcon from '@material-ui/icons/Restore';
+import swal from 'sweetalert';
+const constants = require("../util/constants");
+const { Modal } = require("react-bootstrap/lib");
+let db = require(`${__dirname}/../util/data-provider`).targetDb()
+
+
+export default function BookEditList({ show }) {
```
[the remark this answers is not in the document]
Comment (Ashersam) [reply src/components/BookEditList.js]: Changed Component Name

File: src/components/AutographaStore.js [hunk 4]
```diff
@@ -11,26 +11,27 @@ export class AutographaStore {
...
-  @observable contentCommon =''
-  // @observable contentThree =''
+  @observable contentOne = ''
+  @observable contentTwo = ''
+  @observable contentCommon = ''
   @observable bookActive = 1;
   @observable chapterActive = 1
   @observable currentRef = 'eng_ult'
   @observable bookData = Constant.booksList
+  @observable editBookData = null
```
[the remark this answers is not in the document]
Comment (Ashersam) [reply src/components/AutographaStore.js]: Changed

File: src/components/AutographaStore.js [hunk 5]
```diff
@@ -11,26 +11,27 @@ export class AutographaStore {
...
+  @observable openEditBook = false
```
[the remark this answers is not in the document]
Comment (Ashersam) [reply src/components/AutographaStore.js]: Done with this name change

File: src/components/AutographaStore.js [hunk 6]
```diff
@@ -11,26 +11,27 @@ export class AutographaStore {
...
+  @observable UpdatedBookName = ''
+  @observable editMode = false
```
[the remark this answers is not in the document]
Comment (Ashersam) [reply src/components/AutographaStore.js]: Changed mode name

File: src/components/AutographaStore.js [hunk 7]
```diff
@@ -11,26 +11,27 @@ export class AutographaStore {
...
+  @observable editPopup = false
```
[the remark this answers is not in the document]
Comment (Ashersam) [reply src/components/AutographaStore.js]: Changed popup name

File: src/components/BookEditList.js [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+                    <span>
+                        <TextField
+                            style={{ marginRight: "10px" }}
+                            value={AutographaStore.bookName}
+                            name="defaultValue"
+                            id="defaultValue"
+                        />
+                        <ArrowRightAltIcon />
+                        <TextField
+                            hintText="Translate BookName"
```
[the remark this answers is not in the document]
Comment (Ashersam) [reply src/components/BookEditList.js]: Screenshot from 2020-01-03 11-20-04

[the remark this answers is not in the document]
Comment (Ashersam) [reply src/components/BookEditList.js]: Done with the change

File: src/components/BookEditList.js [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    return (
+        <div>
+            <Modal
+                show={show}
+                style={{ "top": "52px", "left": "154px", "height": "643px", "position": "fixed" }}
+            >
+                <Modal.Header className="head" >
+                    <Modal.Title>UpdateBookNames</Modal.Title>
```
[the remark this answers is not in the document]
Comment (Ashersam) [reply src/components/BookEditList.js]: Screenshot from 2020-01-03 11-25-47

File: src/components/BookEditList.js [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+                            hintText="Translate BookName"
+                            style={{ marginLeft: "10px" }}
+                            onChange={onChange}
+                            required
+                            value={updatedValue || ""}
+                            name="updatedValue"
+                            id="updatedValue"
+                            maxLength={20}
+                        />
+                        <RaisedButton onClick={resetToDefault}><RestoreIcon /></RaisedButton>
```
[the remark this answers is not in the document]
Comment (Ashersam) [reply src/components/BookEditList.js]: Change to Just icon button. 
Note: Refer to the above png

File: src/components/Navbar.js [hunk 11]
```diff
@@ -211,51 +218,84 @@ class Navbar extends React.Component {
...
             swal(currentTrans["dynamic-msg-error"], currentTrans["dynamic-msg-enter-translation"], "error");
-        });  
+        });
     }
 
     openpopupAboutUs() {
         AutographaStore.showModalAboutUs = true
     }
+    componentDidMount() {
+        db.get('001', function (err, doc) {
```
[the remark this answers is not in the document]
Comment (Ashersam) [reply src/components/Navbar.js]: Changed to **translatedBookNames** as mentioned above

**Comment by joelthe1** [comment]
Translations need to updated in the app for the new strings.

**Comment by Ashersam** [comment]
> Translations need to updated in the app for the new strings.

added all translation
Screenshot from 2020-01-06 16-50-55

**Pull request merged**